packages/sdk/src/charge.ts

```diff
@@ -328,11 +328,16 @@ export class Charge {
       throw new Error('getInfo() has not been called');
     }
 
-    await this.initCharge(payerData);
-
-    await this.readyForSettlement();
-
-    await this.submitTransactionOnChain();
+    try {
+      await this.initCharge(payerData);
+      await this.readyForSettlement();
+      await this.submitTransactionOnChain();
+    } catch (e) {
+      if (e instanceof OnchainFailureError) {
+        await this.abort(e.code);
+        throw e;
```

Comment:
Since the `try` is wrapping all three server requests we should still throw the error even if it's not an `OnchainFailureError`

Reply:
Or you can just wrap the `submitTransactionOnChain` with the `try` and then just let the other two throw as they would normally.